client/services/FeedbackGeneratorService.js

```diff
@@ -373,21 +447,58 @@ tie.factory('FeedbackGeneratorService', [
        * @returns {Feedback}
        */
       getFeedback: function(tasks, codeEvalResult, rawCodeLineIndexes) {
+        // Reset all counters but consecutiveSameErrorCounter.
+        _resetCounters('consecutiveSameError');
+
+        // If the user recieves the same error message increment the same error
+        // counter.
+        if (previousErrorString === codeEvalResult.getErrorString()) {
+          consecutiveSameErrorCounter++;
+        }
+
         var feedback = _getFeedbackWithoutReinforcement(
           tasks, codeEvalResult, rawCodeLineIndexes);
         if (tasks.length > 0 && !codeEvalResult.getErrorString()) {
           feedback.setReinforcement(
             ReinforcementGeneratorService.getReinforcement(
               tasks[tasks.length - 1], codeEvalResult));
         }
+
+        // If the same error counter reaches the threshold, prompt the user to
+        // look at the primer.
+        if (consecutiveSameErrorCounter === UNFAMILIARITY_THRESHOLD) {
+          feedback.appendTextParagraph(_getUnfamiliarLanguageFeedback());
+          consecutiveSameErrorCounter = 0;
+        }
+        previousErrorString = codeEvalResult.getErrorString();
         return feedback;
       },
       getSyntaxErrorFeedback: function(errorString) {
+        // Reset all counters but consecutivesyntaxErrorCounter.
+        _resetCounters('consecutiveSyntaxError');
+
+        // If the user recieves another syntax error, increment the syntax
+        // error counter.
+        consecutiveSyntaxErrorCounter++;
         var feedback = FeedbackObjectFactory.create(false);
         feedback.appendSyntaxErrorParagraph(errorString);
+
+        // If the syntax error reaches the threshold, prompt the user to
+        // look at the primer.
+        if (consecutiveSyntaxErrorCounter === UNFAMILIARITY_THRESHOLD) {
+          feedback.appendTextParagraph(_getUnfamiliarLanguageFeedback());
+          consecutiveSyntaxErrorCounter = 0;
```

Comment:
Elena and I talked about this, and we thought it might be best not to keep bugging them to look at the primer if they elect not to. The message will still be present in the feedback history so they can look at the primer whenever they so choose, but we didn't want the constant reminders to be annoying.